tapdb/assets_store.go

```diff
@@ -886,8 +886,13 @@ func (a *AssetStore) constraintsToDbFilter(
 		assetFilter.AssetIDFilter = assetIDBytes
 		assetFilter.KeyGroupFilter = groupKeyBytes
 
-		// TODO(roasbeef): only want to allow asset ID or other and not
-		// both?
+		// If we query by group key, we don't also include the asset ID,
+		// otherwise we'd only get assets from that specific tranche.
+		if query.DistinctSpecifier &&
+			len(assetFilter.KeyGroupFilter) > 0 {
+
+			assetFilter.AssetIDFilter = nil
```

Comment:
Should add a test case for this, assuming we don't go with one of the other options I mentioned in my other comment.

Reply:
Added unit tests. Had to simplify and refactor the `TestFetchAllAssets` test a bit, but should be more easy to add more test cases in the future.